OpenSim/Simulation/SimbodyEngine/BallJoint.h

```diff
@@ -52,14 +52,36 @@ class OSIMSIMULATION_API BallJoint : public Joint {
     OpenSim_DECLARE_CONCRETE_OBJECT(BallJoint, Joint);
 
     /** Specify the Coordinates of the BallJoint */
-    CoordinateIndex rx{ constructCoordinate(Coordinate::MotionType::Rotational) };
-    CoordinateIndex ry{ constructCoordinate(Coordinate::MotionType::Rotational) };
-    CoordinateIndex rz{ constructCoordinate(Coordinate::MotionType::Rotational) };
+    CoordinateIndex rx{constructCoordinate(Coordinate::MotionType::Rotational)};
+    CoordinateIndex ry{constructCoordinate(Coordinate::MotionType::Rotational)};
+    CoordinateIndex rz{constructCoordinate(Coordinate::MotionType::Rotational)};
 
 public:
     /** Use Joint's constructors. @see Joint */
     using Joint::Joint;
 
+    using Joint::getCoordinate;
+
+    /** Indices of Coordinates; can be used with getCoordinate() for
+        convenience. */
+    enum class Coord {
+        Rotation1,
+        Rotation2,
+        Rotation3
+    };
+
+    /** Get the Coordinates associated with this Joint. */
+    const Coordinate& getCoordinate(Coord idx) const {
```

Comment:
> Do you think this method should be on the base `Joint` class (they could be virtual, and implemented by the subclasses).

We were trying to accomplish a couple things with this design. First, it should not be possible to do something like
```cpp
myBallJoint.getCoordinate(PlanarJoint::SomeEnum)
```
For example, a user could change the joint type without changing the argument to `getCoordinate()` and end up with the wrong index into the CoordinateSet. Secondly, it doesn't seem like the base Joint class should know anything about the derived classes, so including all the enums in Joint seems undesirable.

Methods like `BallJoint::getCoordinateR1()` would work as well, but I thought it would be most elegant to have a single `getCoordinate()` that was common to all joint types.